src/subscriptions/SubscriptionManager.cpp

```diff
@@ -388,9 +286,6 @@ SubscriptionManager::pubBookChanges(
     ripple::LedgerInfo const& lgrInfo,
     std::vector<Backend::TransactionAndMetadata> const& transactions)
 {
-    if (bookChangesSubscribers_.empty())
```

Comment:
We should not check the subscribers here. The sub method run asynchronously. If a message need to be published just after subscribing, the session will be ignored.